src/components/ui/Chat/Message/index.tsx

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { HTMLAttributes } from 'react';
+
+import { BaseProps } from '../../Base';
+import { StyledMessage, StyledMessageInfo, StyledMessageContent } from './Styled';
+import PopOver from '../../PopOver';
+import PopOverItem, { PopOverItemProps } from '../../PopOver/PopOverItem';
+import { ChatCaret, Dots } from '../../icons';
+
+export type MessageVariant = 'outgoing' | 'incoming';
+
+export interface MessageProps extends Omit<HTMLAttributes<HTMLDivElement>, 'css'>, BaseProps {
+    /* determines styling for outgoing and incoming messages. */
+    variant: MessageVariant;
+    /* the name of the user that sent the message. */
+    name: string;
+    /* The timestamp the message was created. */
+    time: string;
+    /* The text content of the message. */
+    content: string;
+    /* The options available in the more options button. */
+    moreOptions?: PopOverItemProps[] | null;
+    /* Adds the bubble caret to a message. */
+    showCaret?: boolean;
+}
+
+const Message = (props: MessageProps) => {
+
+    return (
+        <StyledMessage {...props} data-testid='chat-message'>
+            <StyledMessageContent {...props}>
+                <div className='ch-sender-name'>{props.name}</div>
+                <div>{props.content}</div>
+                {props.showCaret && <ChatCaret height='9px' width='4px' />}
+            </StyledMessageContent>
+            <StyledMessageInfo>
```

Comment:
If `moreOptions` is an optional prop, the `PopOver` that renders them should be conditionally rendered based on `moreOptions` being non-null. I can think of use cases where someone might want to build a chat experience w/o including a menu.

Reply:
Great call out. I'll update.